ff/src/fields/arithmetic.rs

```diff
@@ -58,10 +58,10 @@ macro_rules! impl_field_mul_assign {
 }
 
 macro_rules! impl_field_into_repr {
-    ($limbs:expr, $BigIntegerType:ty) => {
+    ($limbs:expr) => {
         #[inline]
-        #[ark_ff_asm::unroll_for_loops]
-        fn into_repr(&self) -> $BigIntegerType {
+        // #[ark_ff_asm::unroll_for_loops]
```

Comment:
Do you feel like the compiler can still optimize this well? Or should we try to do some manual unrolling inside the macro?

Reply:
Well, for now, since we can no longer take advantage of literals in macro instantiations, I've decided one should have match arms and do the loop unroll for each of the possible num limbs. 

This is extremely ugly, and I'm not sure if there's a workaround; the only plausible workaround for unrolling loops parametrised by const is proposal for compiler hints #[optimize(..)] that is not moving atm.

Reply:
So, basically, this is not the final form

Reply:
One can make the actual impl very compact, but under the hood, with macro expand etc it will be very ugly, just like our asm code selecting the num_limbs match arms

Reply:
I think I will simply modify the proc macro, exposing a new one that creates the match arms up to some limit, which can be a meta attr that can be parsed as usize for the match limit.

Reply:
I think, for a specific problem, which is copying the data back over in each of the match arms, since there would be a type mismatch, it might make sense to do an unsafe transmute of a pointer; however, this would violate our desire to have safe code.